blueoil/generate_lmnet_config.py

```diff
@@ -155,9 +163,11 @@ def _blueoil_to_lmnet(blueoil_config):
     learning_rate_func = None
     learning_rate_kwargs = None
     if learning_rate_schedule == "constant":
-        optimizer_kwargs = {"momentum": 0.9, "learning_rate": initial_learning_rate}
+        if optimizer == "Momentum":
```

Comment:
`"momentum": 0.9` should be set when `optimizer` is `Mometum`, 
But `"momentum": 0.9` doesn't be set when `learning_rate_schedule` is not constant, I think.
Please check 👍

Reply:
You are right! I will fix it.

Reply:
Fixed and run blueoil_test.sh and run Adam + (constant / decay) and run Momentum + (constant / decay).

Reply:
When I use this config, `learning_rate_schedule` is `2-step-decay`
```
# supported task types are 'classification', 'object_detection' and 'semantic_segmentation'.
task_type: classification

network_name: LmnetV1Quantize

dataset:
  format: Caltech101
  train_path: ./lmnet/tests/fixtures/datasets/dummy_classification
  test_path: 

trainer:
  batch_size: 1
  epochs: 4
  # supported 'optimizer' is 'Momentum', 'Adam' currently.
  # Momentum
  #    https://www.tensorflow.org/api_docs/python/tf/train/MomentumOptimizer
  # Adam
  #    https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer
  optimizer: Adam
  # supported 'learning_rate_schedule' is 'constant', '2-step-decay', '3-step-decay', '3-step-decay-with-warmup' ({epochs} is the number of training epochs you entered before).
  #   'constant' -> constant learning rate.
  #   '2-step-decay' -> learning rate decrease by 1/10 on {epochs}/2 and {epochs}-1.
  #   '3-step-decay' -> learning rate decrease by 1/10 on {epochs}/3 and {epochs}*2/3 and {epochs}-1.
  #   '3-step-decay-with-warmup' -> warmup learning rate 1/1000 in first epoch, then train the same way as '3-step-decay'.
  learning_rate_schedule: 2-step-decay
  initial_learning_rate: 0.001

network:
  quantize_first_convolution: yes

common:
  image_size:
    - 128  # height
    - 128  # width

  # set pretrain model name. currently, this feature is not supported, always ignored.
  pretrain_model: false

  # enable dataset prefetch, set false if weired problem happens
  dataset_prefetch: true

  data_augmentation:
    - Blur:
        - value: (0, 1)
```

But, Unfortunately, The learning rate is constant.

<img width="369" alt="screen shot 2019-02-06 at 23 23 53" src="https://user-images.githubusercontent.com/138891/52347904-64f02000-2a66-11e9-8be2-6591e7c03768.png">


Need to fix 👍

Reply:
I confirmed with same settings.
The result is OK.

I think no fix is needed.

![optimizer](https://user-images.githubusercontent.com/43975451/52384240-f93aa100-2abf-11e9-8ddf-ca5f676e5fdf.png)

Reply:
Sorry. The root of the problem is deep.
According to @nlpng 's help, setting both piecewise_constant and initilal_learning_value are not compatible.
I will try and confirm four(all) patterns.

Reply:
I confirmed four patterns and seems to work well. Please review again.